units/helpers_windows.go

```diff
@@ -2,12 +2,14 @@ package units
 
 import (
 	"context"
-	"errors"
 	"io"
 	"time"
+
+	"github.com/sirupsen/logrus"
 )
 
 // ReadJournalOutputSince returns error since windows does not support journal
 func ReadJournalOutputSince(ctx context.Context, unit string, duration time.Duration) (io.ReadCloser, error) {
-	return nil, errors.New("there is no journal on Windows")
+	logrus.Warnf("There is no journal for %s on Windows", unit)
+	return nil, nil
```

Comment:
This will most definitely break calling code as it's very idiomatic in Go to expect a non-nil response when the error is nil. It would be better to return a `ioutil.NopCloser(bytes.NewReader(nil))` here.

Reply:
example that would fail: https://github.com/dcos/dcos-diagnostics/blob/a83b9b5141605ce04029dbf43f331c5bee4756ff/api/rest/bundle_handler.go#L213 (I guess our test coverage here is not very good).

Reply:
Good catch!

Reply:
Yep. I just found that the diagnostic bundle creation is not working with this patch.